internal/resources/policy/kind/mutation/recipe/label_flatten_test.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package recipe
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	policyrecipecustomcommonmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/policy/recipe/custom/common"
+	policyrecipemutationmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/policy/recipe/mutation"
+	policyrecipemutationcommonmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/policy/recipe/mutation/common"
+)
+
+func TestFlattenLabel(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		description string
+		input       *policyrecipemutationmodel.VmwareTanzuManageV1alpha1CommonPolicySpecMutationV1Label
+		expected    []interface{}
+	}{
+		{
+			description: "check for nil mutation label",
+		},
+		{
+			description: "flatten normal label mutation policy struct",
+			input: &policyrecipemutationmodel.VmwareTanzuManageV1alpha1CommonPolicySpecMutationV1Label{
+				Label: &policyrecipemutationcommonmodel.KeyValue{
+					Key:   "key_value",
+					Value: "value_value"},
+				Scope: policyrecipemutationcommonmodel.NewVmwareTanzuManageV1alpha1CommonPolicySpecMutationV1Scope(policyrecipemutationcommonmodel.Cluster),
+				TargetKubernetesResources: []*policyrecipecustomcommonmodel.VmwareTanzuManageV1alpha1CommonPolicySpecCustomV1TargetKubernetesResources{
+					{
+						APIGroups: []string{"policy"},
+						Kinds:     []string{"pod"},
+					},
+				},
+			},
+			expected: []interface{}{
+				map[string]interface{}{
+					LabelKey: []interface{}{
```

Comment:
Is `LabelKey` an array of maps here? Please correct me if I'm wrong.

Reply:
Almost, slice containing a single map. Quite common approach in this project.